### DISCUSSION TITLE
Is there any way to break the formatter's context/translator state?

### DISCUSSION THREAD

**Discussion by user202729** [opensteno/plover#1348] [Q&A and Support]
For example, if `R` is mapped to `{#right}` (or `{#right}{}{^}`, `*UPD` is mapped to `{*<}` (or `{:retro_case:upper_first_word}`) and `WORD` is mapped to `word`. Then when I stroke

```
WORD/R/WORD/*UPD
```

I expect Plover to output

```
<word> + {#right} + <word> + {#backspace} *4 + <WORD>
```

Currently the last stroke deletes all the 8 characters (possibly more) and turn all those to uppercase.

What can I do so it doesn't happen?

~~*I just developed a simple plugin that calls [`clear_translator_state`](https://plover2.readthedocs.io/en/latest/api/engine.html#plover.engine.StenoEngine.clear_translator_state),*~~ (**edit**: `clear_translator_state` has some bad side effects) *but is there an existing way?  (`{^ ^}{#backspace}` doesn't count)*

**Comment by antistic** [comment]
> `clear_translator_state` has some bad side effects

What kind of side effects?

> `{^ ^}{#backspace}` doesn't count

I'll second this. It's not ideal because of the [common issue of some programs skipping keys](https://github.com/openstenoproject/plover/issues/1175), or because it actually produces output (not ideal for programs that do something on space key).

**Reply by user202729** [reply]
It has been too long, I think "it doesn't work if used in the middle of a translation." or something like that.

See also https://github.com/openstenoproject/plover_plugins_registry/pull/19 (someone else used that approach and pushed a package)
